File: app/src/main/java/top/someones/cardmatch/ui/ModLiveData.java
package top.someones.cardmatch.ui;

import androidx.lifecycle.LiveData;
import top.someones.cardmatch.entity.Mod;

public class ModLiveData extends LiveData<Mod[]> {

    private static final ModLiveData mModLiveData = new ModLiveData();

    private ModLiveData() {
    }

    public static ModLiveData getLiveData() {
        return mModLiveData;
    }

    @Override
    public void postValue(Mod[] value) {
        super.postValue(value);
    }

    @Override
    public void setValue(Mod[] value) {
        super.setValue(value);
    }

}

File: app/src/main/java/top/someones/cardmatch/ui/ModLiveData.kt
package top.someones.cardmatch.ui

import androidx.lifecycle.LiveData
import top.someones.cardmatch.entity.Mod

class ModLiveData private constructor() : LiveData<Array<Mod>?>() {
    public override fun postValue(value: Array<Mod>?) {
        super.postValue(value)
    }

    public override fun setValue(value: Array<Mod>?) {
        super.setValue(value)
    }

    companion object {
        val liveData = ModLiveData()
    }
}